vowpalwabbit/lda_core.cc

```diff
@@ -1234,7 +1234,7 @@ LEARNER::base_learner *lda_setup(vw &all)
 
   ld.decay_levels.push_back(0.f);
 
-  LEARNER::learner<lda> &l = init_learner(&ld, ld.compute_coherence_metrics ? learn_with_metrics : learn, 1 << all.weights.stride_shift());
+  LEARNER::learner<lda> &l = init_learner(&ld, ld.compute_coherence_metrics ? learn_with_metrics : learn, 1 << all.weights.stride_shift(), prediction_type::scalars);
```

Comment:
The LDA module was returning the wrong prediction type because of this.  @JohnLangford do you think it's a good idea to force all callers of `init_learner` to supply the appropriate `pred_type`?  This would force this problem to come up at compile time instead of relying on the fact that the base learner's type is the same as `this`'s type (which is clearly not true for all cases).